UnityGLTF/Assets/UnityGLTF/Scripts/Extensions/GLTFSceneExporter.Data.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEngine;
+using GLTF.Schema;
+using VectorTypes;
+
+namespace UnityGLTF
+{
+	public partial class GLTFSceneExporter
+	{
+
+		private AccessorId ExportData(GLTFAccessorAttributeType type, GLTFComponentType componentType, int componentSize, int count, IEnumerable<object> min, IEnumerable<object> max, int byteLength, Action<BinaryWriter> writeData, string accessorName = null)
+		{
+			// The offset of the data must be aligned to a multiple of the component size.
+			var position = checked((int)this._bufferWriter.BaseStream.Position);
+			var alignedPosition = Align(position, componentSize);
+			for (var i = position; i < alignedPosition; i++)
+			{
+				this._bufferWriter.Write(byte.MinValue);
+			}
+
+			var bufferViewId = this.ExportBufferView(alignedPosition, byteLength);
+			var accessorId = this.ExportAccessor(bufferViewId, componentType, count, type, min, max, accessorName);
+
+			writeData(this._bufferWriter);
+
+			return accessorId;
+		}
+
+		private AccessorId ExportAccessor(BufferViewId bufferViewId, GLTFComponentType componentType, int count, GLTFAccessorAttributeType type, IEnumerable<object> min, IEnumerable<object> max, string name = null)
+		{
+			Accessor accessor = new Accessor
+			{
+				BufferView = bufferViewId,
+				ComponentType = componentType,
+				Count = count,
+				Type = type,
+				Min = (List<double>) min,
+				Max = (List<double>) max,
+				Name = name
+			};
+
+			var id = new AccessorId
+			{
+				Id = _root.Accessors.Count,
+				Root = _root
+			};
+
+			_root.Accessors.Add(accessor);
+
+			return id;
+		}
+
+		private static int Align(int value, int size)
+		{
+			var remainder = value % size;
+			return (remainder == 0 ? value : checked(value + size - remainder));
```

Comment:
use (value + size - 1) / size * size instead. It will do the alignment without the branch